barretenberg/cpp/src/barretenberg/commitment_schemes/ipa/ipa.hpp

```diff
@@ -742,8 +880,11 @@ template <typename Curve_> class IPA {
      * @return Polynomial<bb::fq> 
      */
     static Polynomial<bb::fq> create_challenge_poly(const size_t log_poly_length_1, const std::vector<bb::fq>& u_challenges_inv_1, const size_t log_poly_length_2, const std::vector<bb::fq>& u_challenges_inv_2, bb::fq alpha) {
-        Polynomial challenge_poly = construct_poly_from_u_challenges_inv(log_poly_length_1, u_challenges_inv_1);
+        // Always extend each to 1<<CONST_ECCVM_LOG_N length
+        Polynomial<bb::fq> challenge_poly(1<<CONST_ECCVM_LOG_N);
```

Comment:
this is slightly inefficient since we += the challenge_poly_1 instead of set challenge_poly's values to challenge_poly_1's values